test/autorest/generated/lrogroup/pollers.go

```diff
@@ -23,25 +25,71 @@ type lrOSCustomHeaderPost202Retry200Poller struct {
 	// the client for making the request
 	client *lrOSCustomHeaderOperations
 	// polling tracker
-	// pt *pollingTrackerPost
+	pt *pollingTrackerPost
 }
 
+// Done returns true if the polling operation has terminated either in a success case or failure case,
+// otherwise it will return false
 func (p *lrOSCustomHeaderPost202Retry200Poller) Done() bool {
-	return false
+	return p.pt.hasTerminated()
 }
 
+// ID ...
 func (p *lrOSCustomHeaderPost202Retry200Poller) ID() string {
-	return "NYI"
+	return ""
 }
 
 func (p *lrOSCustomHeaderPost202Retry200Poller) Poll(ctx context.Context) (*LrOSCustomHeaderPost202Retry200Response, error) {
-	return nil, nil
+	done, err := p.doneWithContext(ctx)
+	if done {
```

Comment:
Remember your happy line.
```go
if done, err := p.doneWithContext(ctx); !done {
    return nil, err
}
// remaining code...
```

Reply:
So I've got a problem here and I have a note on my local version about this. What should be done when done() returns an error? I want to return immediately. Which is what I am going to push. 
So it would end up something like this: 

```
if done, err := p.doneWithContext(ctx); !done || err != nil {
    return nil, err
}
// success case here without errors
```
What do you think? I believe this does vary from what the guidelines are saying

Reply:
I'll merge this for now. But let me know what you think and I'll update my next PR accordingly.

Reply:
The condition works however won't `done()` always return `false` in the event that its returned error is not nil?

Reply:
Probably, but not necessarily. This is the code we have in done()

```
func (p *lrOSCustomHeaderPost202Retry200Poller) done(ctx context.Context) (done bool, err error) {
	if p.pt.hasTerminated() {
		return true, p.pt.pollingError()
	}
	if err := p.pt.pollForStatus(ctx, p.client.p); err != nil {
		return false, err
	}
	if err := p.pt.checkForErrors(); err != nil {
		return p.pt.hasTerminated(), err
	}
	if err := p.pt.updatePollingState(p.pt.provisioningStateApplicable()); err != nil {
		return false, err
	}
	if err := p.pt.initPollingMethod(); err != nil {
		return false, err
	}
	if err := p.pt.updatePollingMethod(); err != nil {
		return false, err
	}
	return p.pt.hasTerminated(), p.pt.pollingError()
}
```

Reply:
Because if polling fails with an error from the service, terminated will be true but there will also be an error.